File: examples/with-nightmare/test/screenshot.test.js
import Nightmare from 'nightmare'
import path from 'path'

describe('nightmare', () => {
  it('should screenshot', async () => {
    const nightmare = new Nightmare()

    await nightmare
      .goto('https://github.com/argos-ci/argos/')
      .screenshot(path.join(__dirname, '../screenshots/screenshot.png'))
      .end()
  })
})

File: examples/with-pytest/test/screenshot_test.py
import os
import time

def take_screenshot(selenium, name):
  filename = '{folder}/{name}.png'.format(
    folder=os.path.join(os.path.dirname(__file__), '../screenshots'),
    name=name
  )
  selenium.get_screenshot_as_file(filename)

def test_example(selenium):
  # Some documentation
  # https://selenium-python.readthedocs.io/getting-started.html
  selenium.get('https://github.com/argos-ci/argos')
  time.sleep(2)
  take_screenshot(selenium, 'homepage')
